### ISSUE TITLE
Crash when attempting to change tags on an azurerm_virtual_machine_scale_set

### ISSUE THREAD

**Issue by cberner** [hashicorp/terraform#13959]
### Terraform Version
0.9.3

### Affected Resource(s)
- azurerm_virtual_machine_scale_set

### Terraform Configuration Files
```hcl
resource "azurerm_resource_group" "testrg" {
  name = "cberner-testrg"
  location = "eastus"
}

resource "azurerm_virtual_network" "test" {
  name                = "acctvn"
  address_space       = ["10.0.0.0/16"]
  location            = "${azurerm_resource_group.testrg.location}"
  resource_group_name = "${azurerm_resource_group.testrg.name}"
}

resource "azurerm_subnet" "test" {
  name                 = "acctsub"
  resource_group_name  = "${azurerm_resource_group.testrg.name}"
  virtual_network_name = "${azurerm_virtual_network.test.name}"
  address_prefix       = "10.0.2.0/24"
}

resource "azurerm_storage_account" "testsa" {
  name                = "randomstring18hfjas983"
  resource_group_name = "${azurerm_resource_group.testrg.name}"

  location     = "eastus"
  account_type = "Standard_LRS"
}

resource "azurerm_storage_container" "test" {
  name                  = "vhds"
  resource_group_name   = "${azurerm_resource_group.testrg.name}"
  storage_account_name  = "${azurerm_storage_account.testsa.name}"
  container_access_type = "private"
}

resource "azurerm_virtual_machine_scale_set" "test" {
  name = "test_ss"
  location = "${azurerm_resource_group.testrg.location}"
  resource_group_name = "${azurerm_resource_group.testrg.name}"
  upgrade_policy_mode = "Manual"

  tags {
    test_tag = "value2"
  }

  sku {
    name = "Standard_D1_v2"
    tier = "Standard"
    capacity = 0
  }

  os_profile {
    computer_name_prefix = "test"
    admin_username = "ubuntu"
    admin_password = "Secret1234!"
  }

  os_profile_linux_config {
    disable_password_authentication = true

    ssh_keys {
      path = "/home/ubuntu/.ssh/authorized_keys"
      key_data = "${file("id_rsa.pub")}"
    }
  }

  storage_profile_image_reference {
     publisher = "Canonical"
     offer     = "UbuntuServer"
     sku       = "14.04.2-LTS"
     version   = "latest"
   }
 
   storage_profile_os_disk {
     caching           = "ReadWrite"
     create_option     = "FromImage"
     name = "osdisktest"
     vhd_containers = ["${azurerm_storage_account.testsa.primary_blob_endpoint}${azurerm_storage_container.test.name}"]
   }

   network_profile {
     name = "nic"
     primary = true

     ip_configuration {
       name = "primary"
       subnet_id = "${azurerm_subnet.test.id}"
    }
  }

  lifecycle {
    # Ideally, we would just ignore sku.capacity, storage_profile_os_disk.image, and os_profile.custom_data
    # but that doesn't seem to work
    ignore_changes = ["sku", "storage_profile_image_reference"]
  }
}

```

### Expected Behavior
Tags should have been updated without crashing

### Actual Behavior
Terraform crashed

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
2. Edit the tags of the Scale Set
3. `terraform apply`

**Comment by grubernaut** [comment]
Hi @cberner, can you please post the output of `crash.log`, either in a gist or a comment? Thanks.

**Comment by cberner** [comment]
Unfortunately no, `crash.log` appears to contain sensitive authorization information. If there's a piece of it that you need I can certainly post that

**Comment by grubernaut** [comment]
@cberner, that's exactly the piece we need, thanks.

**Comment by grubernaut** [comment]
@cberner this was fixed in #13464, and should be released in v0.9.4 shortly. Thanks!
